app/src/modules/settings/routes/extensions/extensions.vue

```diff
@@ -1,25 +1,41 @@
 <script setup lang="ts">
 import api from '@/api';
-import { ApiOutput, EXTENSION_TYPES } from '@directus/extensions';
+import { APP_OR_HYBRID_EXTENSION_TYPES, ApiOutput, ExtensionType } from '@directus/extensions';
 import { groupBy } from 'lodash';
 import { computed, ref } from 'vue';
 import { useI18n } from 'vue-i18n';
+import { useRouter } from 'vue-router';
 import SettingsNavigation from '../../components/navigation.vue';
 import ExtensionGroupDivider from './components/extension-group-divider.vue';
 import ExtensionItem from './components/extension-item.vue';
 import ExtensionsInfoSidebarDetail from './components/extensions-info-sidebar-detail.vue';
+import VNotice from '@/components/v-notice.vue';
+import { useReloadGuard } from '@/composables/use-reload-guard';
 
 const { t } = useI18n();
+const router = useRouter();
 
 const error = ref();
 const loading = ref(false);
 const extensions = ref<ApiOutput[]>([]);
+const needsReload = ref(false);
 
 const bundled = computed(() => extensions.value.filter(({ bundle }) => !!bundle));
-const regular = computed(() => extensions.value.filter(({ bundle }) => !!bundle === false));
-
+const regular = computed(() => extensions.value.filter(({ bundle }) => !bundle));
 const extensionsByType = computed(() => groupBy(regular.value, 'schema.type'));
 
+const { confirmLeave, leaveTo, removeGuard } = useReloadGuard(needsReload);
+
+const currentPageLink = () => document.location.href;
+
+const leavePage = () => {
+	removeGuard();
+	// push target location into the route history
+	router.push(leaveTo.value ?? currentPageLink());
+	// hard-reload the page using the latest route
+	router.go(0);
+};
```

Comment:
Instead of `removeGuard` we need to set `needsReload` to false, otherwise the guard still blocks the `router.push` call and `router.go` results in reloading the current page.
Note: The event listener should be removed anyway since a) the component is unmounted b) the page is hard-reloaded.

Also we should await `router.push` in this case to ensure `router.go` is called after navigation is completed.

```suggestion
const currentPageLink = document.location.href;

const leavePage = async () => {
	needsReload.value = false;
	// push target location into the route history
	await router.push(leaveTo.value ?? currentPageLink);
	// hard-reload the page using the latest route
	router.go(0);
};
```

Reply:
I found a way to resolve the url instead so we can simply rely on `document.location.href` for navigating instead of this vue router workaround.